src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/CacheEntry.java

```diff
@@ -15,35 +15,588 @@
+        }
+        return lazyInitiatedResponseStaleAt;
+    }
+
+    private Instant responseReceivedAt()
+    {
+        if (lazyInitiatedResponseReceivedAt == null)
+        {
+            final ListFW<HttpHeaderFW> responseHeaders = getCachedResponseHeaders();
+            final String dateHeaderValue = getHeader(responseHeaders, "date") != null ?
+                    getHeader(responseHeaders, "date") : getHeader(responseHeaders, "last-modified");
+            try
+            {
+                Date receivedDate = DATE_FORMAT.parse(dateHeaderValue);
+                lazyInitiatedResponseReceivedAt = receivedDate.toInstant();
+            }
+            catch (Exception e)
+            {
+                lazyInitiatedResponseReceivedAt = Instant.EPOCH;
+            }
+        }
+        return lazyInitiatedResponseReceivedAt;
+    }
+
+
+    private CacheControl responseCacheControl()
+    {
+        ListFW<HttpHeaderFW> responseHeaders = getCachedResponseHeaders();
+        String cacheControl = getHeader(responseHeaders, CACHE_CONTROL);
+        return cache.responseCacheControlFW.parse(cacheControl);
+    }
+
+
+    public boolean canServeRequest(
+        ListFW<HttpHeaderFW> request,
+        short authScope)
+    {
+        if (this.state == CacheEntryState.PURGED)
+        {
+            return false;
+        }
+        Instant now = Instant.now();
+
+        final boolean canBeServedToAuthorized = canBeServedToAuthorized(request, authScope);
+        final boolean doesNotVaryBy = doesNotVaryBy(request);
+        final boolean satisfiesFreshnessRequirements = satisfiesFreshnessRequirementsOf(request, now);
+        final boolean satisfiesStalenessRequirements = satisfiesStalenessRequirementsOf(request, now)
+                /*|| this.state == CAN_REFRESH */|| this.state == REFRESHING;
+        final boolean satisfiesAgeRequirements = satisfiesAgeRequirementsOf(request, now);
+        return canBeServedToAuthorized &&
+                doesNotVaryBy &&
+                satisfiesFreshnessRequirements &&
+                satisfiesStalenessRequirements &&
+                satisfiesAgeRequirements;
+    }
+
+    private boolean isStale()
+    {
+        return Instant.now().isAfter(staleAt());
+    }
+
+    protected boolean isIntendedForSingleUser()
+    {
+        ListFW<HttpHeaderFW> responseHeaders = getCachedResponseHeaders();
+        if (SurrogateControl.isXProtected(responseHeaders))
+        {
+            return false;
+        }
+        else
+        {
+            // TODO pull out as utility of CacheUtils
+            String cacheControl = HttpHeadersUtil.getHeader(responseHeaders, "cache-control");
+            if (cacheControl == null)
+            {
+                return false;
+            }
+            final CacheControl parsedCacheControl = cache.responseCacheControlFW.parse(cacheControl);
+            return parsedCacheControl.contains("private");
```

Comment:
`return cacheControl != null && cache.responseCacheControlFW.parse(cacheControl).contains("private")`

Also use constant for `"private"`.